lib/src/validator/sdk_constraint.dart

```diff
@@ -58,6 +46,21 @@ class SdkConstraintValidator extends Validator {
             'See https://dart.dev/tools/pub/publishing#publishing-prereleases '
             'For more information on pre-releases.');
       }
+      if (
+          // We only want to give this hint if there was no other problems with
+          // the sdk constraint.
+          warnings.isEmpty &&
+              errors.isEmpty &&
+              originalConstraint != effectiveConstraint) {
```

Comment:
This will only happen in Dart 3.x, right? and this change won't make it into 2.x?

Reply:
Context: I don't think we want people to write `<4.0.0` before they are using a 3.0.0-... SDK.

Reply:
Yes, unless we cherry-pick it to the 2.19-series. The sdk main branch is already in dart 3-land.